src/odemis/dataio/catmaid.py

```diff
@@ -0,0 +1,343 @@
+         base_url (str): URL where the Catmaid instance is hosted.
+         project_id (int): project id, indicating which of the projects to use at this Catmaid instance.
+         stack_id (int): stack id, indicating which of the stacks to use at this Catmaid instance.
+         return:
+             data (DataArrayShadows): DataArrayShadows representing the image
+         """
+        stack_info = get_stack_info(base_url, project_id, stack_id)
+        # Dimension of the image volume in pixels at its original scale level.
+        data = [DataArrayShadowPyramidalCatmaid(stack_info, base_url)]
+        return data
+
+
+TILE_URLS = {
+    FILE_BASED: '{image_base}{depth}/{row}_{col}_{zoom}.{file_extension}',
+    FILE_BASED_WITH_ZOOM_DIRS: '{image_base}{depth}/{zoom}/{row}_{col}.{file_extension}',
+    DIR_BASED: '{image_base}{zoom}/{depth}/{row}/{col}.{file_extension}',
+    RENDER_SERVICE: '{image_base}largeDataTileSource/{tile_width}/{tile_height}/{zoom}/'
+                    '{depth}/{row}/{col}.{file_extension}',
+    FLIXSERVER: '{image_base}{depth}/{row}_{col}_{zoom}.{file_extension}',
+}
+
+
+def format_tile_url(tile_source_type, image_base, zoom, depth, row, col, file_extension, tile_width, tile_height):
+    """
+    Format a URL that can be used to request a tile from the server.
+    https://catmaid.readthedocs.io/en/stable/tile_sources.html
+
+    tile_source_type (int): A number indicating which tile source type to use for this project.
+    image_base (str): URL where the images are located.
+    zoom (0<=int): zoom level to use.
+    depth (0<=int): The Z index of the stack.
+    row (0<=int): Y index of the tile.
+    col (0<=int): X index of the tile.
+    file_extension (str): File extension of the image.
+    tile_width (int): Width of the tiles in pixels.
+    tile_height (int): Height of the tiles in pixels.
+    return:
+        tile_url (str): URL where the tile is located.
+    """
+
+    try:
+        tile_url = TILE_URLS[tile_source_type]
+    except KeyError:
+        raise LookupError("Tile Source Type {} is not supported".format(tile_source_type))
+    return tile_url.format(image_base=image_base, zoom=zoom, depth=depth, row=row, col=col,
+                           file_extension=file_extension, tile_width=tile_width, tile_height=tile_height)
+
+
+def response_to_array(response):
+    """
+    response (Response): http response for the requested image.
+    return:
+       image (numpy array): the requested image from the response.
+    """
+    response.raise_for_status()
+    content_type = response.headers['Content-Type']
+
+    if content_type in SUPPORTED_CONTENT_TYPES:
+        buffer = BytesIO(response.content)  # opening directly from raw response doesn't work for JPEGs
+        raw_img = Image.open(buffer)
+        return numpy.array(raw_img)
+    else:
+        logging.info('Image fetching is only implemented for greyscale PNG and JPEG, not {}'.format(
```

Comment:
It needs to raise a `IOError` or `ValueError` in such case. returning a None when the data type is not supported is just the best way to make the error hard to find when it happens.